File: LogWriterVB/Excepcion.vb
Public Class Excepcion
    Inherits Exception
    Private Const _mensaje_predeterminado As String = "Ha ocurrido algo, contacte con el soporte técnico."

    Public Sub New()
        MyBase.New(_mensaje_predeterminado)
    End Sub

    Public Sub New(message As String)
        MyBase.New(message)
    End Sub

    Public Sub New(message As String, inner As Exception)
        MyBase.New(message, inner)
    End Sub
End Class

File: LogWriterCS/Excepcion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    class Excepcion : Exception
    {
        private const string _mensaje_predeterminado = "Ha ocurrido algo, por favor contacte con el soporte técnico.";
        
        public Excepcion() : base(_mensaje_predeterminado)
        {
        }

        public Excepcion(string message) : base(message)
        {
        }

        public Excepcion(string message, Exception inner) : base(message, inner)
        {
        }
    }
}
